<!DOCTYPE html>
<html>
<head>
    <title> Finally</title>
    <style>
        body {
            background-color: #1a1a1a;
            color: white;
            font-family: Arial, sans-serif;
            font-size: 16px;
        }

        h1 {
            text-align: center;
        }

        .card {
            background-color: #262626;
            border-radius: 10px;
            box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.3);
            margin: 20px auto;
            max-width: 600px;
            padding: 20px;
        }

        pre {
            background-color: #333333;
            border-radius: 5px;
            color: lightgreen;
            padding: 10px;
            overflow-x: scroll;
        }

        .output {
            background-color: #333333;
            border-radius: 5px;
            color: white;
            padding: 10px;
        }

        .output p {
            margin: 0;
        }

        #theory{
            background-color: #262626;
            border-radius: 10px;
            box-shadow: 0px 2px 5px rgba(255, 255, 255, 0.3);
            margin: 20px auto;
            max-width: 1500px;
            padding: 20px;
            padding: 10px;
            border-radius: 5px;
            font-size: x-large;
            color: violet;

        }
    </style>
</head>
<body>
<h1>Java Finally Keyword</h1>
<div id = "theory"> In Java, the "finally" keyword is used in conjunction with try-catch blocks
    to define a block of code that will be executed regardless of whether an exception is thrown or not. The "finally" block is placed after the "catch" block(s) and is used to specify a
    set of statements that must be executed before the control flow leaves the try-catch block. </div>
<div class="card">
    <h2>Example</h2>
    <pre>
public class TestFinallyBlock2{
    public static void main(String args[]){

        try {

        System.out.println("Inside try block");

        //below code throws divide by zero exception
        int data=25/0;
        System.out.println(data);
        }

        //handles the Arithmetic Exception / Divide by zero exception
            catch(ArithmeticException e){
                System.out.println("Exception handled");
                    System.out.println(e);
        }

        //executes regardless of exception occured or not
        finally {
                System.out.println("finally block is always executed");
        }

                System.out.println("rest of the code...");
        }
}
    </pre>
    <h3>Output:</h3>
    <div class="output">
        <p>Index out of bounds!</p>
        <p>Inside the finally block</p>
    </div>
</div>
</body>
</html>